pkgs/development/compilers/dtc/0001-Depend-on-.a-instead-of-.so-when-building-static.patch

```diff
@@ -0,0 +1,101 @@
+From aa244c17846318357d2edfb418e174916aa95e78 Mon Sep 17 00:00:00 2001
+From: Tero Tervala <tero.tervala@unikie.com>
+Date: Wed, 15 Jun 2022 13:44:55 +0300
+Subject: [PATCH] Depend on .a instead of .so when building static
+
+Static build may be requested by setting environment variable:
+STATIC_BUILD=1
+
+Checks are skipped on static builds
+
+Signed-off-by: Tero Tervala <tero.tervala@unikie.com>
+---
+ Makefile             | 16 +++++++++++++---
+ tests/Makefile.tests | 11 ++++++++---
+ 2 files changed, 21 insertions(+), 6 deletions(-)
+
+diff --git a/Makefile b/Makefile
+index ee77115..87b7990 100644
+--- a/Makefile
++++ b/Makefile
+@@ -25,6 +25,10 @@ WARNINGS = -Wall -Wpointer-arith -Wcast-qual -Wnested-externs -Wsign-compare \
+ 	-Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wshadow
+ CFLAGS = -g -Os $(SHAREDLIB_CFLAGS) -Werror $(WARNINGS) $(EXTRA_CFLAGS)
+ 
++ifeq ($(STATIC_BUILD),1)
++	CFLAGS += -static
++endif
++
```

Comment:
I can still build `pkgsStatic.dtc` without this change.  Is it necessary?

Reply:
It's not absolutely necessary when -static is already given in EXTRA_CFLAGS, which seems to be the case when building pkgsStatic.dtc

Reply:
This part removed